pkg/netns/netns_linux.go

```diff
@@ -0,0 +1,213 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package netns
+
+import (
+	"fmt"
+	"os"
+	"runtime"
+
+	"golang.org/x/sync/errgroup"
+	"golang.org/x/sys/unix"
+)
+
+type NetNS struct {
+	fd int
+}
+
+// CreateNewNetns creates a network namespace and returns a handle to it.
+//
+// The current thread will not be moved to this new network namespace as with a basic
+// UNSHARE syscall - the original network namespace will be retained and a handle to the
+// new network namespace will be returned.
+//
+// The namespace created by this call is NOT pinned, or bind-mounted, and so will be
+// closed/lost when the process terminates.
+func New() (*NetNS, error) {
+	var innerFD int
+
+	g := new(errgroup.Group)
+	// Force this into a dedicated + threadlocked goroutine to avoid disturbing the caller's
+	// thread.
+	g.Go(func() error {
+		restore, err := saveRestoreCurrentThreadNetNS()
+		if err != nil {
+			return fmt.Errorf("Unable to obtain and lock current netns: %w", err)
+		}
+		defer restore()
```

Comment:
Yeah I guess since we can rely on the runtime to reap the thread if we don't unlock it, we have an "out" if the defer somehow doesn't run, so they're safe to drop.